### PR TITLE
add check_permutation.rs

### PR DESCRIPTION
**Pull request by mengfansheng-git**
Given two strings s1 and s2.
Determine whether the characters of one of the strings can be rearranged to become another string.

### CODE REVIEW

File: src/string/check_permutation.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+//! Given two strings s1 and s2 consisting of lowercase letters.
```
Comment (vil02) [inline src/string/check_permutation.rs:1]: What if they are not lowercase letters?

File: src/string/check_permutation.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+//!
+//! Determine whether the characters of one of the strings can be
+//! rearranged to become another string.
+
+pub fn check_permutation(s1: &str, s2: &str) -> bool {
+    if s1.len() != s2.len() {
+        return false;
+    }
+    s1.chars()
+        .all(|c| s1.matches(c).count() == s2.matches(c).count())
```
Comment (vil02) [inline src/string/check_permutation.rs:11]: What is the time complexity of this? $O(n^2)$? I think one can do it in a _linear_ time.

**Review by vil02 — CHANGES_REQUESTED** [review]

File: src/string/check_permutation.rs (hunk 1 above)
Comment (mengfansheng-git) [reply src/string/check_permutation.rs:1]: have no effect

File: src/string/check_permutation.rs (hunk 2 above)
Comment (mengfansheng-git) [reply src/string/check_permutation.rs:11]: With the help of a hash table to record the number of occurrences of a character and then compare the time complexity should be linear time.

**Pull request closed without merging**